l3build/l3build.lua

```diff
@@ -2125,20 +2125,21 @@ bundleunpack = bundleunpack or function(sourcedir)
     end
   end
   for _,i in ipairs(unpackfiles) do
-    for _,j in ipairs(filelist(unpackdir, i)) do
+    for j,_ in pairs(tree(unpackdir, i)) do
       -- This 'yes' business is needed to pass a series of "y\n" to
       -- TeX if \askforoverwrite is true
       -- That is all done using a file as it's the only way on Windows and
       -- on Unix the "yes" command can't be used inside execute (it never
       -- stops, which confuses Lua)
       execute(os_yes .. ">>" .. localdir .. "/yes")
+      local path, name = splitpath(j)
       local localdir = abspath(localdir)
       errorlevel = run(
-        unpackdir,
+        unpackdir .. "/" .. path,
```

Comment:
Yes, the core intention is that unpacked files should retain the same directory structure as their sources are assorted in the project layout. I came to this conclusion by thinking about the following structure.

    .
    \-- document.tex
    \-- dir
        \-- include.tex

In `document.tex`, `include.tex` is used in compiled form. Then, in `document.tex`, I suspect it to be more intuitive to write

    \includegraphics{dir/include.pdf}

than

    \includegraphics{include.pdf}

It also avoids name clashes when files in different directories have the same names.